apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -141,31 +144,42 @@ private void publishEvent(
       Partition partition,
       Partition oldpartition)
     throws MetaException {
-    JSONObject json = createBaseMessage(eventType, table.getDbName(), table.getTableName());
+    String sourceMetastoreUris = table.getParameters().get(METASTOREURIS.varname);
```

Comment:
@abhimanyugupta07 in apiary we will have more than one metastore, fronted by network load balancer and route53 dns. 

your application should consume events from a SNS topic and use corresponding DNS/NLB to connect to metastore.

so SNS topic ARN & metastore dns should be used in circustrain config.

Reply:
Is `table.getParameters().get(METASTOREURIS.varname)` the right place to get this? I was expecting to get it from the Hive conf somewhere (perhaps going via `new HiveConf()`) but I guess either way @rpoluri's comment still stands that it will say something like `localhost` which won't be of much use downstream in Shunting Yard. So I can think of two possible solutions, one is that we configure this Apiary listener with the Metastore URIs and pull it out of the listener config (similar to what we do with the ARN). Or we have this config on the Shunting Yard side. @rpoluri where do you think this will be easier to set up and maintain?

Reply:
OK, the only downside is that we would then have to keep this in sync in two places - Apiary and Shunting Yard while having it be part of Apiary would mean we could set it one place. Apiary should know what it's end points are. But I see your point in that where people end up consuming the event may mean they have completely different ways of getting to Apiary that Apiary can't predict?

Reply:
we are registering dns entries, so will have predictable names, hence will not need to update consumer configs even if we redeploy datalake.

usually following 2 entires should be in consumer configs.
SNS_ARN: arn:aws:sns:{aws_region}:{aws_account}:apiary-{instance}-metadata-events
Metastore_URI: thrift://hms-readwrite.apiary-{instance}-{aws_region}.lcl

Reply:
that metastore name mentioned is only valid for internal clients...

if someone is consuming sns events from outside the account, those metastore uris will not be valid.
and we will be adding this metastore uri to every event... which seems unnecessary as that will not change between all events from that topic.

If you still wan't we can add that as option, by I strongly feel this leads to strange situations, backend service may not know all entry points and shouldn't try to publish for callback.

Don't we have to configure destination metastore URI in shunting yard, can't we specify source metastore in same context? and SNS will be just a trigger to push data from source to destination?

Reply:
OK, so it sounds like it will be impossible to predict the external names here. @abhimanyugupta07 can you remove the setting of the metastoreUris element from the PR and action the other changes please?

Reply:
Alright. I have made the relevant changes.

Reply:
I have made the changes. Preserving the order turned out to be easier than I expected. I just needed to pass in LinkedHashMap to the JSONObject and it is preserving the order. Also, I realize that the project is not using org.json's JSON library but this one: https://github.com/tdunning/open-json which uses whatever kind of map is passed as argument.